test/common/network/utility_test.cc

```diff
@@ -28,6 +33,41 @@ namespace Envoy {
 namespace Network {
 namespace {
 
+struct Interface {
+  std::string name;
+  uint32_t if_index;
+};
+
+// Helper function that returns any usable interface present while running the test.
+#ifndef WIN32
+StatusOr<Interface> getLocalNetworkInterface() {
+  struct ifaddrs* ifaddr;
+
+  if (getifaddrs(&ifaddr) == -1) {
+    return absl::FailedPreconditionError("getifaddrs failed");
+  }
+
+  // Loop on all the interfaces and try to find one with IPv6 address.
+  for (struct ifaddrs* ifa = ifaddr; ifa != nullptr; ifa = ifa->ifa_next) {
+    if (ifa->ifa_addr == nullptr) {
```

Comment:
I don't think it is necessary just because the getaddrinfo and the if_nametoindex just care about the interface existence so any interface should be good.

Reply:
Updated the misleading comment that was referring to looking for an IPv6 address.